addons/middle-click-popup/userscript.js

```diff
@@ -0,0 +1,469 @@
+            let len = split[iM].length;
+            bText.appendChild(document.createTextNode(procCode.substr(i, len)));
+            li.appendChild(bText);
+            i += len;
+          }
+
+          if (i < procCode.length) {
+            li.appendChild(document.createTextNode(procCode.substr(i)));
+          }
+
+          if (count === 0) {
+            li.classList.add("sel");
+          } else {
+            li.classList.remove("sel");
+          }
+          count++;
+        } else {
+          li.style.display = "none";
+          li.classList.remove("sel");
+        }
+      }
+      p.append(this.dropdown);
+    }
+
+    inputKeyDown(e) {
+      if (e.keyCode === 38) {
+        this.navigateFloatFilter(-1);
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 40) {
+        this.navigateFloatFilter(1);
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 13) {
+        // Enter
+        let sel = this.dropdown.querySelector(".sel");
+        if (sel) {
+          this.onClick(e);
+          this.hide();
+        }
+        e.cancelBubble = true;
+        e.preventDefault();
+        return;
+      }
+      if (e.keyCode === 27) {
+        // Escape
+        if (this.floatInput.value.length > 0) {
+          this.floatInput.value = ""; // Clear search first, then close on second press
+          this.inputChange(e);
+        } else {
+          this.hide();
+        }
+        e.preventDefault();
+        return;
+      }
+    }
+
+    buildFilterList() {
+      let options = [];
+
+      let t = this.workspace.getToolbox();
+
+      let blocks = t.flyout_.getWorkspace().getTopBlocks();
+      // 107 blocks, not in order... but we can sort by y value or description right :)
+
+      let fullDom = Blockly.Xml.workspaceToDom(t.flyout_.getWorkspace());
+      const doms = {};
+      for (const x of fullDom.children) {
+        if (x.tagName === "BLOCK") {
+          let id = x.getAttribute("id");
+          doms[id] = x;
+        }
+      }
+
+      for (const block of blocks) {
+        this.getBlockText(block, options, doms);
+      }
+
+      options.sort((a, b) =>
+        a.desc.length < b.desc.length ? -1 : a.desc.length > b.desc.length ? 1 : a.desc.localeCompare(b.desc)
+      );
+
+      let count = 0;
+
+      while (this.dropdown.firstChild) {
+        this.dropdown.removeChild(this.dropdown.firstChild);
+      }
+      for (const option of options) {
+        const li = document.createElement("li");
+        const desc = option.desc;
+
+        // bType = hat block reporter boolean
+
+        let bType = this.getEdgeTypeClass(option.block);
+
+        count++;
+
+        li.innerText = desc;
+        li.data = { text: desc, lower: " " + desc.toLowerCase(), option: option };
+
+        li.className =
+          "sa-block-color-" + (option.block.isScratchExtension ? "pen" : option.block.getCategory()) + " sa-" + bType;
```

Comment:
`block.isScratchExtension` doesn't seem to work anymore, I think you can use something like `addon.tab.getCustomBlock(option.block.mutation.proccode)` instead